### PR TITLE
[ENG-1172] Integrate scripts

### PR DESCRIPTION
**Pull request by corbinSanders**
## Purpose

Pulling in all scripts and creating a jupyter script to run everything together.

## Changes

Making small changes to scripts to make them work together better.
Creating ia_workflow.ipynb as a notebook to run all five scripts

## QA Notes

Ensure all scripts run. See dev notes below. 

## Documentation

N/A

## Side Effects

N/A

## DevNotes
Setup a virtualenv to run this. Ensure you've placed your OSF token, Datacite credentials, and IA credentials in the settings files. Run requirements with `pip3 install -r requirements.txt`. Then run Jupyter with `jupyter notebook ia_workflow.ipynb`.

## Ticket

https://openscience.atlassian.net/browse/ENG-1172

### CODE REVIEW

File: IA/IA_consume_files.py [hunk 1]
```diff
@@ -55,16 +51,7 @@ def consume_files(guid, token, directory, retry=True):
 
     try:
         response = requests.get(zip_url.format(guid), headers=auth_header)
-        if response.status_code == 429:
-            keep_trying = retry
```
Comment (brianjgeiger) [inline IA/IA_consume_files.py:59]: Why did you remove all the retry logic?

**Review by brianjgeiger — COMMENTED** [review]
Overall, looks good, but I'm wondering why you removed the WB retry logic. Technically it's not supported *yet*, but it will be one day, and it would be nice to handle it out of the gate.

File: settings/defaults.py [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
 OSF_COLLECTION_NAME = 'cos-dev-sandbox'
+OSF_BEARER_TOKEN = ''
 
-IA_ACCESS_KEY = 'change to valid token'
-IA_SECRET_KEY = 'change to valid token'
+IA_ACCESS_KEY = ''
+IA_SECRET_KEY = ''
 
 OSF_API_URL = 'https://localhost:8000/'
+OSF_FILES_URL = 'https://localhost:8000/'
```
Comment (Johnetordoff) [inline settings/defaults.py]: This is the same as `OSF_API_URL` and incorrect, the actual "files" url is waterbutler at `https://localhost:7777/`

File: IA/IA_consume_files.py (hunk 1 above)
Comment (corbinSanders) [reply IA/IA_consume_files.py:59]: I removed it because it is not yet supported, but it would be a good idea to have it future-ready.

File: settings/defaults.py [hunk 3]
```diff
@@ -1,10 +1,22 @@
...
+
+DATACITE_USERNAME = ''
+DATACITE_PASSWORD = ''
+DATACITE_URL = 'https://doi.test.datacite.org/'
+DATACITE_PREFIX = '10.70102'  # Datacite's test DOI prefix -- update in production
+
 CHUNK_SIZE = 1000
-OSF_API_URI = 'http://localhost:8000/'
+PAGE_SIZE = 100
+OSF_API_URI = 'https://localhost:8000/'
```
Comment (brianjgeiger) [inline settings/defaults.py]:
```suggestion
```

Comment (brianjgeiger) [reply settings/defaults.py]: Looks like this [isn't being used in the codebase](https://github.com/CenterForOpenScience/osf-pigeon/search?q=OSF_API_URI&unscoped_q=OSF_API_URI) anywhere.

File: IA/IA_consume_files.py [hunk 4]
```diff
@@ -53,25 +49,27 @@ def consume_files(guid, token, directory, retry=True):
...
-                logging.log(logging.ERROR, 'Throttled. Please retry after {wait_time}s')
-        elif response.status_code >= 400:
-            status_code = response.status_code
-            content = getattr(response, 'content', None)
-            raise requests.exceptions.HTTPError(
-                'Status code {}. {}'.format(status_code, content))
-    except requests.exceptions.RequestException as e:
-        logging.log(logging.ERROR, 'HTTP Request failed: {}'.format(e))
-        raise
+            if response.status_code >= 400:
```
Comment (brianjgeiger) [inline IA/IA_consume_files.py]:
```suggestion
            elif response.status_code >= 400:
```

Comment (brianjgeiger) [reply IA/IA_consume_files.py]: This looks like it'll raise an exception if we get a 429.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
A couple of small things, otherwise looks great.

**Comment by brianjgeiger** [comment]
Also looks like `flake8` isn't happy.

File: IA/IA_consume_files.py (hunk 1 above)
Comment (Johnetordoff) [reply IA/IA_consume_files.py:59]: could you use the the `get_with_retry` function in `utils.py`?

Comment (corbinSanders) [reply IA/IA_consume_files.py:59]: I had to modify it to utilize headers, but as you wish.

**Review by brianjgeiger — APPROVED** [review]
🎉 🐧 🎉
